CityPuzzle/Login And SingUp/SignUpPage.xaml.cs

```diff
@@ -21,70 +18,47 @@ public SignUpPage()
         {
             InitializeComponent();
         }
- 
-        static Regex Valid_Username = StringNumber();
-        static Regex Valid_Contact = NumbersOnly();
-        static Regex Valid_Password = ValidPassword();
-        static Regex Valid_Email = Email_Address();
-
-        private static Regex StringNumber()
-        {
-            string StringAndNumber_Pattern = "^[a-zA-Z][a-zA-Z0-9]{5,11}";
-
-            return new Regex(StringAndNumber_Pattern, RegexOptions.IgnoreCase);
-        }
 
+        public delegate bool Validator(string text);
 
         public static void Validation(List<Entry> fields)
         {
+            Validator validateUsername = new Validator(RegexDelegate.validUsername);
+            Validator validatePassword = new Validator(RegexDelegate.validPassword);
+            Validator validateEmail = new Validator(RegexDelegate.validEmail);
+
             foreach (Entry field in fields)
             {
-                if (String.IsNullOrWhiteSpace(field.Text)) throw new EmptyInputdException(field.Placeholder, field);
+                if (String.IsNullOrWhiteSpace(field.Text))
+                    throw new EmptyInputdException(field.Placeholder, field);
 
+                // field.Placeholder is defined in SignUpPage.xaml Entity propertie for text fields
                 switch (field.Placeholder)
                 {
                     case "Vartotojo vardas":
-                        if (!Valid_Username.IsMatch(field.Text)) throw new BadInputdException("nuo 6 iki 12 ilgio", field.Placeholder, field);
-                        else if (!User.CheckUser(field.Text)) throw new BadInputdException("Naudotojas su tokiu prisijungimo vardu užimtas.Pakeiskite prisijungimo vardą ir bandykite dar kartą.", field);
+                        if (!validateUsername(field.Text)) 
+                            throw new BadInputdException("nuo 6 iki 12 ilgio", field.Placeholder, field);
+                        else if (!User.CheckUser(field.Text)) 
+                            throw new BadInputdException("Naudotojas su tokiu prisijungimo vardu užimtas.Pakeiskite prisijungimo vardą ir bandykite dar kartą.", field);
+                        break;
+                    case "Slaptazodis":
+                        if (!validatePassword(field.Text)) 
+                            throw new BadInputdException("Password must be atleast 8 to 15 characters and contains atleast one Upper case and numbers.", field);
```

Comment:
Error messages need editing; should be in one language.

Reply:
Updated all messages to be in Lithuanian